typings/index.d.ts

```diff
@@ -182,6 +183,18 @@ export abstract class AnonymousGuild extends BaseGuild {
   public splashURL(options?: StaticImageURLOptions): string | null;
 }
 
+export class APIRequest {
+  private constructor(rest: unknown, method: string, path: string, options: unknown);
+  public client: Client;
+  public method: string;
```

Comment:
This could be a union of these: https://github.com/discordjs/discord.js/blob/994abf509e6b10b536c23f928f100307f2dc0109/src/rest/APIRouter.js#L4

Reply:
That's not *strictly* true though. Someone could do the unthinkable and call `client.rest.request('options', ...` for example. I don't necessarily disagree, though because that's an unsupported use case I suppose.

Reply:
I remember also making a PR to node-fetch to specify the types of methods and they said they supported all methods, case insensitive, so this should still remain as string imo

Reply:
`string` can still be added to a union while preserving autocompletion through a type.

```ts
type LiteralUnion<T> = T | (string & { _?: never });

declare const method: LiteralUnion<'get' | 'post' | 'delete' | 'patch' | 'put'>;
```

Reply:
I mostly think it would be okay to call out the methods directly here without the string type since you'd have to be using a private property / method already so you'd already be ignoring ts.